src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1209,8 +1226,19 @@ else if (!referencePath.isDirectory())
                     String sModulePath = getSubmodulePath(sModuleName);
 
                     perModuleArgs.add(sModulePath);
-                    launchCommandWithCredentials(perModuleArgs, workspace, cred, urIish, timeout);
+                    StandardCredentials finalCred = cred;
+                    URIish finalUrIish = urIish;
+                    executorService.submit(() -> {
```

Comment:
This change has a big issue - now exceptions are swallowed. This should really be fixed.

Reply:
We're only catching `InterruptedException`, and even then, we re-throw it as `GitException`. Perhaps I'm misunderstanding something, but I don't believe that this change is swallowing all exceptions.

Reply:
You can verify the described behaviour by writing a test that performs a git submodule call with an invalid parameter. No output is logged and no exception is thrown.

Have a look at the [Future<?> submit(Runnable task)](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html#submit-java.lang.Runnable-). The returned `Future` has a [V get()](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html#get--) method which throws an `ExecutionException` if the computation threw an exception.

Every `Future` should be checked for an exception.

Some useful references:
* https://www.baeldung.com/java-executor-service-tutorial
* https://www.baeldung.com/java-runnable-callable
* https://www.baeldung.com/java-executor-wait-for-threads

Reply:
@darxriggs Thanks for the additional information. I'll try to put together a patch in a new PR (along with a test) to fix this behavior.